tasks/docker.py

```diff
@@ -21,3 +27,70 @@ def integration_tests(ctx, skip_image_build=False, skip_build=False):
     print("Starting docker integration tests")
     env = {"DOCKER_IMAGE": DOGSTATSD_TAG}
     ctx.run("./test/integration/docker/dsd_alpine_listening.sh", env=env)
+
+
+@task
+def dockerize_test(ctx, binary, skip_cleanup=False):
+    """
+    Run a go test in a remote docker environment and pipe its output to stdout
+    """
+    import docker
+
+    client = docker.from_env()
+    temp_folder = tempfile.mkdtemp(prefix="ddtest-")
+
+    ctx.run("cp %s %s/test.bin" % (binary, temp_folder))
+
+    with open("%s/Dockerfile" % temp_folder, 'w') as stream:
```

Comment:
This part is a little sketchy, I'm wondering if we should not have a real DockerFile in `test/integration` ?

Reply:
As dockerfiles don't support conditionnal layers, I don't see how to optionnaly add the testdata folder if it exists. We'd need to open the dockerfile in python to add the file anyway.
But I aggree with you, that's not great for readability, I'll investigate how to improve on that.